pypeit/pypeit.py

```diff
@@ -858,19 +858,26 @@ def extract_one(self, frames, det, sciImg, objFind, initial_sky, sobjs_obj):
             basename=self.basename)
 
         if not self.par['reduce']['extraction']['skip_extraction']:
-            skymodel, objmodel, ivarmodel, outmask, sobjs, waveImg, \
-                tilts = self.exTract.run(final_global_sky, ra=self.fitstbl["ra"][frames[0]],
-                                         dec=self.fitstbl["dec"][frames[0]], obstime=self.obstime)
+            # Perform the extraction
+            self.exTract.run(final_global_sky)
```

Comment:
Please don't change this. I prefer that run return values rather than just assigning form the class.

Reply:
Let's discuss @rcooke-ast. I think a control flow where there are return values is much more clear. My suggestion is that if you run without actually extracting that you returning None.

Reply:
@rcooke-ast I agree that we need to discuss this in general in pypeit but it needs to be clear what the run method returns. For that reason I'm asking that you not modify run to not return variables. After you've changed that back, I'll approve. If you are running self.run differently in the IFU child such that run does not return anything then: 1)  I'd suggest maybe splitting up what run does into multiple methods and only run the sub-method that you need for IFU, 2) returning None in places where you don't need what run returns. I don't have the top-level view of how IFU using this class, but I'm very much against the object oriented approach we've adopted everywhere in PypeIt and trying to retain some semblance of functional programming in places is very important to me, which is why I designed many of these methods to (somewhat redudnantly) return variables so it is clear what the control flow is.